File: app/src/main/java/mvp/model/domain/GameDomainContract.java
package mvp.model.domain;

import mvp.model.entity.Card;

/**
 * Created by Manuel Lorenzo
 */

public interface GameDomainContract {
    interface Manager {
        void cardFlipped(int position, Card card);

        void removeCardsFromMap();

        boolean shouldDispatchUiEvent();
    }
}

File: app/src/main/java/kittentrate/ui/game/GameDomainContract.kt
package kittentrate.ui.game

/**
 * Created by Manuel Lorenzo
 */

interface GameDomainContract {
    interface Manager {
        fun cardFlipped(position: Int, card: Card)

        fun removeCardsFromMap()

        fun shouldDispatchUiEvent(): Boolean

        fun resetScore()
    }
}
